plugins/module_utils/mso.py

```diff
@@ -429,73 +435,103 @@ def request_download(self, path, destination=None):
         data = None
```

Comment:
Why are we setting `data = None` here? Should this not occur prior to the if src statement or as part of the else condition?